File: src/wsTripXML/Code/Classes/cLogData.vb
Imports System
Imports TripXMLMain
Imports System.Threading
Imports System.IO.File
Imports System.Xml
Imports System.Text
Imports System.Web.Configuration


Namespace wsTravelTalk


    Public Class cLogData
        Private mstrRequest As String = ""
        Private mstrResponse As String = ""

        Private sb As StringBuilder = New StringBuilder()

        Public Sub LogDataDeals(ByVal strRequest As String, ByVal strResponse As String)

            mstrRequest = strRequest
            mstrResponse = strResponse

            If WebConfigurationManager.AppSettings("DataDatabase") <> Nothing Then
                Dim oLofThread As New Thread(New ThreadStart(AddressOf LogDeals))

                oLofThread.Start()
            End If
            
        End Sub

        Private Sub LogDeals()
            Dim oDA As cDA = Nothing

            Try
                oDA = New cDA("DataDatabase")
                oDA.AddDeals(mstrRequest, mstrResponse)

            Catch ex As Exception
            Finally
                If Not oDA Is Nothing Then
                    oDA.Dispose()
                End If
            End Try

        End Sub

        Public Function GetDataDeals(ByVal strRequest As String) As String
            Dim oDA As cDA = Nothing
            Dim strResponse As String = ""

            Try
                If WebConfigurationManager.AppSettings("DataDatabase") <> Nothing Then
                    oDA = New cDA("DataDatabase")
                    strResponse = oDA.GetDeals(strRequest)
                End If

                Return strResponse

            Catch ex As Exception
            Finally
                If Not oDA Is Nothing Then
                    oDA.Dispose()
                End If
            End Try
            Return String.Empty
        End Function

    End Class

End Namespace

File: src/wsTripXML/Code/Classes/cLogData.cs
using Microsoft.Extensions.Configuration;
using System;
using System.Text;
using System.Threading;
using TripXMLMain;

namespace wsTripXML.wsTravelTalk
{

    public class cLogData
    {
        private string mstrRequest = "";
        private string mstrResponse = "";

        private StringBuilder sb = new StringBuilder();
        private readonly IConfiguration _configuration;

        public cLogData(IConfiguration configuration)
        {
            _configuration = configuration;
        }

        public void LogDataDeals(string strRequest, string strResponse)
        {

            mstrRequest = strRequest;
            mstrResponse = strResponse;

            if (!string.IsNullOrEmpty(_configuration["DataDatabase"]))
            {
                var oLofThread = new Thread(new ThreadStart(LogDeals));

                oLofThread.Start();
            }

        }

        private void LogDeals()
        {
            cDA oDA = null;

            try
            {
                oDA = new cDA("DataDatabase");
                oDA.AddDeals(mstrRequest, mstrResponse);
            }

            catch (Exception ex)
            {
            }
            finally
            {
                if (oDA is not null)
                {
                    oDA.Dispose();
                }
            }

        }

        public string GetDataDeals(string strRequest)
        {
            cDA oDA = null;
            string strResponse = "";

            try
            {
                if (!string.IsNullOrEmpty(_configuration["DataDatabase"]))
                {
                    oDA = new cDA("DataDatabase");
                    strResponse = oDA.GetDeals(strRequest);
                }

                return strResponse;
            }

            catch (Exception ex)
            {
            }
            finally
            {
                if (oDA is not null)
                {
                    oDA.Dispose();
                }
            }
            return string.Empty;
        }

    }

}